agentai/api.py

```diff
@@ -66,10 +72,10 @@ def get_function_arguments(message, conversation: Conversation, function_registr
                 message, conversation, function_registry=function_registry, model=model
             )
         return function_arguments
-    raise ValueError(f"Unexpected message: {message}")
+    raise Exception(f"Unexpected message: {message}")
 
 
-@retry(retry=retry_if_exception_type(ValueError), stop=stop_after_attempt(3))
+@retry_unless_invalid
```

Comment:
Keep the retry decorators separate, we'll have different backoff plans for one pass vs 2 pass functions

Reply:
Done.